### PR TITLE
Deploy agent on metal instances launched for KMT

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
This PR deploys the datadog-agent on the metal instances launched as part of KMT.

### Motivation
Get metrics on the resource utilization of the metal instance. These can then be used to make decisions about changing instance types, and restructuring the tests for more efficient resource usage.

### Additional Notes
Associated PR in [test-infra-definitions](https://github.com/DataDog/test-infra-definitions/pull/442)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by val06 — APPROVED** [review]

File: test/new-e2e/scenarios/system-probe/main.go [hunk 1]
```diff
@@ -52,6 +52,8 @@ func main() {
 	dependenciesDirectoryPtr := flag.String("dependencies-dir", DD_AGENT_TESTING_DIR, "directory where dependencies package is present")
 	vmconfigPathPtr := flag.String("vmconfig", defaultVMConfigPath, "vmconfig path")
 	local := flag.Bool("local", false, "is scenario running locally")
+	runAgentPtr := flag.Bool("run-agent", false, "Run datadog agent on the metal instance")
+	agentVersionPtr := flag.String("agent-version", "7.49", "Version of datadog-agent")
```
Comment (brycekahle) [inline test/new-e2e/scenarios/system-probe/main.go:56]: For a future PR: we should explore having this dynamically set to the latest released version, so we don't have to keep updating it.

File: test/new-e2e/system-probe/system-probe-test-env.go [hunk 2]
```diff
@@ -136,6 +143,10 @@ func getAvailabilityZone(env string, azIndx int) string {
 	return ""
 }
 
+func trueStr(t bool) string {
```
Comment (brycekahle) [inline test/new-e2e/system-probe/system-probe-test-env.go:146]: Use `strconv.FormatBool` instead.

Comment (brycekahle) [reply test/new-e2e/system-probe/system-probe-test-env.go:146]: preferably remove this function, and use `strconv.FormatBool` inline where needed

File: test/new-e2e/go.mod [hunk 3]
```diff
@@ -72,7 +72,7 @@ require (
 	github.com/aws/aws-sdk-go-v2/internal/configsources v1.2.1 // indirect
 	github.com/aws/aws-sdk-go-v2/internal/endpoints/v2 v2.5.1 // indirect
 	github.com/aws/aws-sdk-go-v2/internal/ini v1.3.42 // indirect
-	github.com/aws/aws-sdk-go-v2/service/ecs v1.29.6
+	github.com/aws/aws-sdk-go-v2/service/ecs v1.30.1
```
Comment (usamasaqib) [inline test/new-e2e/go.mod:75]: This is changed to match the version in the go.mod file of test-infra-definitions

**Comment by brycekahle** [comment]
Looks like the setup job failed and a retry doesn't work. Can you ensure KMT passes?

**Comment by usamasaqib** [comment]
> Looks like the setup job failed and a retry doesn't work. Can you ensure KMT passes?

The initial failure is a known condition. 
The retry failed because the underlying metal instance was cleaned up.

I will run the pipeline again to make it green

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform owned file !

**Review by brycekahle — APPROVED** [review]

File: test/new-e2e/scenarios/system-probe/main.go (hunk 1 above)
Comment (KevinFairise2) [reply test/new-e2e/scenarios/system-probe/main.go:56]: FWIW: If you do not specify a version for the agent, the framework should automatically install the latest version of the agent. cf https://github.com/DataDog/test-infra-definitions/blob/main/components/datadog/agentparams/params.go#L56

**Review by KevinFairise2 — APPROVED** [review]

**Pull request merged**